### PR TITLE
Add Op UnitTest for batchnorm

### PR DESCRIPTION
**Pull request by Liyulingyue**
## 描述

From https://github.com/PaddlePaddle/CINN/issues/1378

为Reduce类型算子增加新的测试用例

序号 | 算子名 | 单测文件 | 
-- | -- | -- | 
2| batch_norm| test_batch_norm_op.py | 
3| batch_norm_grad| test_batch_norm_op.py | 

### 算子类型

- [x] ElementWise：输入张量索引和输出张量索引之间存在一对一的对应关系
- [ ] Broadcast：输入张量索引和输出张量索引之间存在一对多的对应关系
- [ ] Injective：单射算子，可以将一个输出 axis 映射到一个输入 axis
- [ ] Reduction：输入张量索引和输出张量索引之间存在多对一的对应关系
- [ ] OutFusible：复杂算子，仍然可以将一对一的算子融合到其输出中。
- [ ] kNonFusible：无法融合的算子

### OpMapper

- [ ] 该算子是否 OpMapper? 如果是，请贴出在 Paddle 中对应的 OpMaker 代码路径。（给出 Github 链接更好）

### 张量维度

- [ ] 1D 张量
- [ ] 2D 张量
- [ ] 3D 张量
- [x] 4D 张量

#### special shape

挑选 2D/3D/4D 张量中的一个，测试下面的特殊情况。

- [x] 其中一个维度为 1
- [x] 其中一个维度小于 1024
- [x] 其中一个维度大于 1024
- [ ] 向量的所有维度都是 1

### 张量数据类型

- [ ] int32
- [ ] int64
- [x] float16
- [x] float32
- [ ] float64

### 广播

- [ ] 这个算子是否支持广播？
- [ ] 广播的测试样例

### 算子属性

算子属性的测试用例。

- [ ] 属性：属性类型-可取值
    - [ ] `op_type`
    - [ ] `axis`
    - [ ] `keepdim`
- [ ] 使用 OpTestHelper 测试上述属性的笛卡尔积组合

### 备注

### CODE REVIEW

**Comment by Liyulingyue** [comment]
@thisjiang @FisherWY

**Review by thisjiang — COMMENTED** [review]
文件有conflict了~另外麻烦用TestCaseHelper吧

File: python/tests/ops/test_batch_norm_op.py [hunk 1]
```diff
@@ -70,20 +65,25 @@ def build_cinn_program(self, target):
...
+    def init_attrs(self):
+        self.class_name = "TestBatchNormTrainOpBase"
+        self.cls = TestBatchNormTrainOp
 
-    def test_check_results(self):
-        self.check_outputs_and_grads(max_relative_error=1e-3)
+        self.inputs = []
+        for x_shape in [[2, 16, 8, 8], [2, 16, 8, 1], [2, 16, 2048, 8]]:
+            for x_type in ["float16", "float32"]:
+                self.inputs.append({
```
Comment (FisherWY) [inline python/tests/ops/test_batch_norm_op.py]: 建议将数据初始化操作放到`init_case`中，该类只负责配置测试的`shape, dtype, attr`，看起来更简洁

**Comment by Liyulingyue** [comment]
@thisjiang @FisherWY

File: python/tests/ops/test_batch_norm_op.py [hunk 2]
```diff
@@ -242,4 +259,5 @@ def test_check_results(self):
 
 
 if __name__ == "__main__":
-    unittest.main()
+    TestBatchNormTrainOpAll().run()
+    TestBatchNormBackwardOpAll().run()
```
Comment (FisherWY) [inline python/tests/ops/test_batch_norm_op.py:286]: 少了`TestBatchNormInferOp`单测的改进，这几个单测的输入可以复用，直接继承一个测试类后修改下`self.class_name`和`self.cls`即可，如：

```python
class TestBatchNormBackwardOpAll(TestBatchNormTrainOpAll):
    def init_attrs(self):
        super().init_attrs()
        self.class_name = "TestBatchNormBackwardOpCase"
        self.cls = TestBatchNormBackwardOp
```

**Comment by FisherWY** [comment]
CINN目前已升级至C++17（[PR1526](https://github.com/PaddlePaddle/CINN/pull/1526)），CI的Docker镜像也已同步更新，可以`merge`一下`develop`分支，防止出现CI失败的问题（`cudnn.so`未找到的问题）

```bash
git fetch origin
git merge origin/develop
```

**Comment by Liyulingyue** [comment]
@thisjiang @FisherWY

**Review by thisjiang — APPROVED** [review]
LGTM
